src/main/java/rx/internal/schedulers/NewThreadWorker.java

```diff
@@ -50,6 +53,16 @@
         PURGE = new AtomicReference<ScheduledExecutorService>();
         PURGE_FORCE = Boolean.getBoolean(PURGE_FORCE_KEY);
         PURGE_FREQUENCY = Integer.getInteger(FREQUENCY_KEY, 1000);
+
+        // Cached to prevent method call
+        IS_ANDROID = PlatformDependent.isAndroid();
```

Comment:
You are checking if the API is `null` which is equivalent (and thus redundant) with this boolean.